Remove SQLite version support caveats [ci skip]

eugeneius opened the pull request:
Spotted when reviewing https://github.com/rails/rails/pull/36240.

Since https://github.com/rails/rails/pull/32923, Active Record requires SQLite version 3.8.0 or greater, so savepoints and partial indexes are always available.

That commit also added a runtime version check, so we can remove the minimum version requirement from the internal adapter documentation.

vipulnsward commented:
We can also update https://github.com/rails/rails/blob/7cc368621524a567781f3a64db135e6bead1c1bb/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb#L738 to not mention the version.

eugeneius commented:
Good 👀 @vipulnsward, updated!

vishaltelangre commented:
I think there is no need to mention the SQLite version here, too.
https://github.com/rails/rails/blob/65f9e0c4bbc56f208cb6b2329d890c6d9935da1d/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L51-L53

eugeneius committed:
Remove SQLite version support caveats [ci skip]

Since d1a74c1e012ed96f7179e53b9190b7da0a369e11, Active Record requires
SQLite version 3.8.0 or greater, so savepoints and partial indexes are
always available.

That commit also added a runtime version check, so we can remove the
minimum version requirement from the internal adapter documentation.

eugeneius commented:
I pushed another update, thanks @vishaltelangre! 🙇
